include/clock_control.h

```diff
@@ -23,23 +24,81 @@ extern "C" {
 /* Used to select all subsystem of a clock controller */
 #define CLOCK_CONTROL_SUBSYS_ALL	NULL
 
+/**
+ * @brief Current clock status.
+ */
+enum clock_control_status {
+	CLOCK_CONTROL_STATUS_OFF,
+	CLOCK_CONTROL_STATUS_STARTING,
```

Comment:
STARTING is the 'most likely' return value of async_on when polling, hence should be of the value 0. CPUs will take less cycles to compare the zero flag.

Reply:
flipped